cli/src/commonMain/kotlin/StartCommand.kt

```diff
@@ -100,15 +113,98 @@ class StartCommand : SuspendingCliktCommand(name = "start") {
         ContextStorage.saveLatestRunId(ortRun.id)
 
         if (wait) {
-            while (ortRun.isRunning()) {
-                delay(POLL_INTERVAL)
-                ortRun = client.repositories.getOrtRun(repositoryId, ortRun.index)
-                if (!useJsonFormat) echoMessage(ortRun.toPrintable())
+            if (useJsonFormat) {
+                while (ortRun.isRunning()) {
+                    delay(POLL_INTERVAL)
+                    ortRun = client.repositories.getOrtRun(repositoryId, ortRun.index)
+                }
+            } else {
+                echoMessage(ortRun.toPrintable())
+                ortRun = updateProgressBar(ortRun, client)
             }
         }
 
         echoMessage(ortRun.toPrintable())
     }
+
+    private fun updateProgressBar(ortRun: OrtRun, client: OrtServerClient): OrtRun = runBlocking {
+        val maxProgressBarValue = JobType.entries.count { it.isConfigured(ortRun.jobConfigs) }.withInitialState()
+
+        val progress = progressBarContextLayout {
+            text { context }
+            spinner(Spinner.Dots())
+            progressBar(width = 30)
+            percentage()
+        }.animateInCoroutine(
+            terminal,
+            context = "Starting",
+            total = maxProgressBarValue,
+            completed = 0
+        )
+
+        launch { progress.execute() }
+
+        var updatedOrtRun = ortRun.copy()
+        while (updatedOrtRun.isRunning()) {
+            delay(POLL_INTERVAL)
+            updatedOrtRun = client.repositories.getOrtRun(repositoryId, updatedOrtRun.index)
+            progress.update {
+                context =
+                    JobType.entries.reversed().find { it.isStarted(updatedOrtRun.jobs) }?.displayName ?: "Starting"
+                completed = JobType.entries.count { it.isFinished(updatedOrtRun.jobs) }.withInitialState()
+            }
+        }
+
+        progress.update {
+            context = "Finished"
+            completed = maxProgressBarValue
+        }
+
+        updatedOrtRun
+    }
+
+    private enum class JobType(val displayName: String) {
+        ANALYZER("Analyzing"),
+        ADVISOR("Advising"),
+        SCANNER("Scanning"),
+        EVALUATOR("Evaluating"),
+        REPORTER("Reporting"),
+        NOTIFIER("Notifying");
+
+        fun isConfigured(config: JobConfigurations) = when (this) {
+            ANALYZER -> true
```

Comment:
`isConfigured`  returns the same thing as `isStarted`. Can't only one function be used ?

Reply:
`isConfigured` returns if a job was configured by the user.
`isStarted` returns if the actual job was started by the ORT Server.

E.g. if the user enables the reporter, it will return `true` for `isConfigured` from the very beginning. But will only be `true` for `isStarted` when the worker actually started.

Reply:
I see, I misread the implementation of the functions.